service/header/store.go

```diff
@@ -14,9 +14,9 @@ import (
 // TODO(@Wondertan): Those values must be configurable and proper defaults should be set for specific node type.
 var (
 	// DefaultStoreCacheSize defines the amount of max entries allowed in the Header Store cache.
-	DefaultStoreCacheSize = 10240
+	DefaultStoreCacheSize = 4096
 	// DefaultIndexCacheSize defines the amount of max entries allowed in the Height to Hash index cache.
-	DefaultIndexCacheSize = 2048
+	DefaultIndexCacheSize = 16384
```

Comment:
It's impossible for a reviewer to know why these numbers are better than others. Instead of repeatedly changing and tweaking these now, we should instead come up with metrics that we want to improve. 
Doesn't require any changes in this PR though.